### PR TITLE
Gradle shadow plugin for dependency resolving

### PR DESCRIPTION
**Pull request by adigyran**
Based on Kitlith change of build.gradle. Using now library for Slime Commons dependency resolving. Changed gradle.yml accordingly. It produces slimevr.jar file

Changed from java library to java application
Based on this #32  with Kitlith's permission

### CODE REVIEW

File: .github/workflows/gradle.yml [hunk 1]
```diff
@@ -3,63 +3,63 @@
...
+        with:
+          repository: Eirenliel/slime-java-commons
+          # Relative path under $GITHUB_WORKSPACE to place the repository
+          path: Slime Java Commons
+
+      - name: Set up JDK 8
+        uses: actions/setup-java@v2.1.0
+        with:
+          java-version: '8'
+          distribution: 'adopt'
```
Comment (ButterscotchVanilla) [inline .github/workflows/gradle.yml:51]:
```suggestion
          distribution: 'adopt'
          cache: 'gradle' # will restore cache of dependencies and wrappers
```

File: .github/workflows/gradle.yml [hunk 2]
```diff
@@ -3,63 +3,63 @@
...
```
Comment (ButterscotchVanilla) [inline .github/workflows/gradle.yml:26]:
```suggestion
          distribution: 'adopt'
          cache: 'gradle' # will restore cache of dependencies and wrappers
```

**Review by ButterscotchVanilla — APPROVED** [review]
Caching can be added to GitHub Actions for gradle

**Comment by adigyran** [comment]
Sorry for amount of commits, we got confused with request changes

**Pull request merged**